Action failing with setup that has previously been working

Getting the following error when running action. Action was working yesterday, and has worked for almost a year.

```
Running self-installer...
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 10 seconds. 2 retries left.
   WARN  GET https://registry.npmjs.org/pnpm error (ERR_INVALID_THIS). Will retry in 1 minute. 1 retries left.
   ERR_PNPM_META_FETCH_FAIL  GET https://registry.npmjs.org/pnpm: Value of "this" must be of type URLSearchParams
Error: Something went wrong, self-installer exits with code 1
Installation Completed!
```

This is our setup:

```
      - name: Install pnpm
        uses: pnpm/action-setup@v2
        with:
          version: latest
```

I change my pnpm/action-setup@v4，but still error。
  Error: Multiple versions of pnpm specified:
  Error: Error: Multiple versions of pnpm specified:
    - version 8.11.0 in the GitHub Action config with the key "version"
    - version pnpm@9.0.6 in the package.json with the key "packageManager"
  Remove one of these versions to avoid version mismatch errors like ERR_PNPM_BAD_PM_VERSION
    - version 8.11.0 in the GitHub Action config with the key "version"
    - version pnpm@9.0.6 in the package.json with the key "packageManager"
  Remove one of these versions to avoid version mismatch errors like ERR_PNPM_BAD_PM_VERSION
      at readTarget (/home/runner/work/_actions/pnpm/action-setup/v4/dist/index.js:1:4528)
      at runSelfInstaller (/home/runner/work/_actions/pnpm/action-setup/v4/dist/index.js:1:3742)
      at async install (/home/runner/work/_actions/pnpm/action-setup/v4/dist/index.js:1:2976)
      at async main (/home/runner/work/_actions/pnpm/action-setup/v4/dist/index.js:1:444)
![image](https://github.com/pnpm/action-setup/assets/149561351/1c60b5e3-20cd-46f0-b920-d28a506e30eb)

--- a/.github/workflows/pr-check-changeset.yml
+++ b/.github/workflows/pr-check-changeset.yml
@@ -28,7 +28,7 @@ jobs:
           ref: ${{ github.event.pull_request.head.sha }} # Check out the head commit, not the merge commit
 
       # install and configure node, pnpm and the changeset tools
-      - uses: pnpm/action-setup@d882d12c64e032187b2edb46d3a0d003b7a43598 # ratchet:pnpm/action-setup@v2
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # ratchet:pnpm/action-setup@v4
       - uses: actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7 # ratchet:actions/setup-node@v3
         with:
           node-version-file: .nvmrc

--- a/.github/workflows/push-tag-create-release.yml
+++ b/.github/workflows/push-tag-create-release.yml
@@ -26,7 +26,7 @@ jobs:
           fetch-depth: "0" # all history
           persist-credentials: false
 
-      - uses: pnpm/action-setup@d882d12c64e032187b2edb46d3a0d003b7a43598 # ratchet:pnpm/action-setup@v2
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # ratchet:pnpm/action-setup@v4
       - uses: actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7 # ratchet:actions/setup-node@v3
         with:
           node-version-file: .nvmrc

--- a/.github/workflows/push-tag-create-release.yml
+++ b/.github/workflows/push-tag-create-release.yml
@@ -12,6 +12,12 @@ on:
   push:
     tags:
       - "*_v*"
+  # Allow manually triggering this workflow from the web UI
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'git release tag to process'
+        required: true
 
 permissions:
   contents: write
@@ -51,9 +57,17 @@ jobs:
           flub --help
           flub commands
 
+      - name: Set tag name from push
+        if: github.event_name == 'push'
+        run: echo "TAG_NAME=${GITHUB_REF}" >> $GITHUB_ENV
+
+      - name: Set tag name from manual input
+        if: github.event_name == 'workflow_dispatch'
+        run: echo "TAG_NAME=refs/tags/${{ github.event.inputs.tag }}" >> $GITHUB_ENV
+
       - name: Get release metadata JSON
         run: |
-          flub release fromTag ${{ github.ref }} --json | jq -c > release-metadata.json
+          flub release fromTag $TAG_NAME --json | jq -c > release-metadata.json
       - name: Upload release metadata JSON
         uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # ratchet:actions/upload-artifact@v3
         with:

--- a/.github/workflows/website-validation.yml
+++ b/.github/workflows/website-validation.yml
@@ -21,7 +21,7 @@ jobs:
       - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # ratchet:actions/checkout@v4
         with:
           submodules: false
-      - uses: pnpm/action-setup@d882d12c64e032187b2edb46d3a0d003b7a43598 # ratchet:pnpm/action-setup@v2
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # ratchet:pnpm/action-setup@v4
       - uses: actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7 # ratchet:actions/setup-node@v3
         with:
           node-version: "18"
@@ -45,7 +45,7 @@ jobs:
       - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # ratchet:actions/checkout@v4
         with:
           submodules: false
-      - uses: pnpm/action-setup@d882d12c64e032187b2edb46d3a0d003b7a43598 # ratchet:pnpm/action-setup@v2
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # ratchet:pnpm/action-setup@v4
       - uses: actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7 # ratchet:actions/setup-node@v3
         with:
           node-version: "18"
@@ -64,7 +64,7 @@ jobs:
       - uses: actions/checkout@f43a0e5ff2bd294095638e18286ca9a3d1956744 # ratchet:actions/checkout@v3
         with:
           submodules: false
-      - uses: pnpm/action-setup@d882d12c64e032187b2edb46d3a0d003b7a43598 # ratchet:pnpm/action-setup@v2
+      - uses: pnpm/action-setup@fe02b34f77f8bc703788d5817da081398fad5dd2 # ratchet:pnpm/action-setup@v4
       - uses: actions/setup-node@1a4442cacd436585916779262731d5b162bc6ec7 # ratchet:actions/setup-node@v3
         with:
           node-version: "18"

